secretstores/azure/keyvault/authutils.go

```diff
@@ -17,15 +17,39 @@ import (
 	"golang.org/x/crypto/pkcs12"
 )
 
-// ClientAuthorizer provides the options to get a bearer authorizer from a client certificate.
-type ClientAuthorizer struct {
+// EnvironmentSettings hold settings to authenticate with the Key Vault.
+type EnvironmentSettings struct {
+	Values map[string]string
+}
+
+// CertConfig provides the options to get a bearer authorizer from a client certificate.
+type CertConfig struct {
 	*auth.ClientCertificateConfig
 	CertificateData []byte
 }
 
-// NewClientAuthorizer creates an ClientAuthorizer object configured to obtain an Authorizer through Client Credentials.
-func NewClientAuthorizer(certificatePath string, certificateBytes []byte, certificatePassword string, clientID string, tenantID string) ClientAuthorizer {
-	return ClientAuthorizer{
+// GetClientCert creates a config object from the available certificate credentials.
+// An error is returned if no certificate credentials are available.
+func (k *keyvaultSecretStore) GetClientCert(settings EnvironmentSettings) (CertConfig, error) {
```

Comment:
No need to make a type member method..this is still coupling.

Reply:
Resolved in fa4a4a0